### PR TITLE
s3-benchmark test with uplink

### PR DESCRIPTION
**Pull request by littleskunk**
At the moment the `uplink cat` has a bug and the benchmark test will error out.

### CODE REVIEW

File: cmd/s3-benchmark/main.go [hunk 1]
```diff
@@ -63,6 +63,8 @@ func main() {
 		client, err = NewMinio(conf)
 	case "aws-cli":
 		client, err = NewAWSCLI(conf)
+	case "uplink":
```
Comment (egonelbre) [inline cmd/s3-benchmark/main.go]: Also add this to help so people know they can use it.

File: cmd/s3-benchmark/uplink.go [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+}
+
+// NewUplink creates new Client
+func NewUplink(conf Config) (Client, error) {
+	return &Uplink{conf}, nil
+}
+
+func (client *Uplink) cmd(subargs ...string) *exec.Cmd {
+	args := []string{
+		"--enc-key", "abc123",
```
Comment (egonelbre) [inline cmd/s3-benchmark/uplink.go]: Use a value that can be specified via a flag.

**Comment by littleskunk** [comment]
@egonelbre Please review it once more. I added `API key` and `satellite address` as well.

File: cmd/s3-benchmark/main.go [hunk 3]
```diff
@@ -23,11 +23,14 @@ func main() {
 	var conf Config
 
 	flag.StringVar(&conf.Endpoint, "endpoint", "127.0.0.1:7777", "endpoint address")
+	flag.StringVar(&conf.SatelliteAddr, "satelliteaddr", "127.0.0.1:7778", "satellite address")
```
Comment (egonelbre) [inline cmd/s3-benchmark/main.go]: Wouldn't this be the same as endpoint? Just having argument as `satellite` would be fine.

File: cmd/s3-benchmark/uplink.go [hunk 4]
```diff
@@ -0,0 +1,130 @@
...
+	conf Config
+}
+
+// NewUplink creates new Client
+func NewUplink(conf Config) (Client, error) {
+	client := &Uplink{conf}
+
+	cmd := client.cmd("setup",
+		"--overwrite",
+		"--api-key="+client.conf.APIKey,
```
Comment (egonelbre) [inline cmd/s3-benchmark/uplink.go]: nit, You can use `"--api-key", client.conf.APIKey,`.

File: cmd/s3-benchmark/main.go (hunk 3 above)
Comment (egonelbre) [reply cmd/s3-benchmark/main.go]: Also might be confusing having both `endpoint` and `satellite`. Maybe rename `endpoint` to `s3-endpoint` or rename such that the same variable covers both?

Comment (littleskunk) [reply cmd/s3-benchmark/main.go]: I was thinking about using `endpoint` for both but it is not the same port. If we have two different options we could call both with the same line.
`s3-benchmark ... -endpoint localhost:7777 -satelliteaddr localhost:7778 -client ...`

Let me rename the options to s3-gateway and satellite.

**Pull request merged**